code/modules/mob/new_player/preferences_setup.dm

```diff
@@ -313,22 +313,56 @@
 
 		preview_icon.Blend(temp, ICON_OVERLAY)
 
-	//Markings
-	if(current_species && ((current_species.bodyflags & HAS_HEAD_MARKINGS) || (current_species.bodyflags & HAS_BODY_MARKINGS)))
```

Comment:
Another redundant if statement broken apart.

Reply:
I also fixed an issue where the code was not checking for `do_colouration` here.